src/find/matchers/samefile.rs

```diff
@@ -0,0 +1,66 @@
+// This file is part of the uutils findutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::fs::Metadata;
+#[cfg(unix)]
+use std::os::unix::fs::MetadataExt;
+
+use super::Matcher;
+
+pub struct SameFileMatcher {
+    metadata: Metadata,
+}
+
+impl SameFileMatcher {
+    pub fn new(metadata: Metadata) -> SameFileMatcher {
+        SameFileMatcher { metadata }
+    }
+}
+
+impl Matcher for SameFileMatcher {
+    #[cfg(unix)]
+    fn matches(&self, file_info: &walkdir::DirEntry, _matcher_io: &mut super::MatcherIO) -> bool {
+        let meta = file_info.metadata().unwrap();
+        meta.ino() == self.metadata.ino()
```

Comment:
You also need to check `meta.dev()`.  Otherwise you'll return true for two unrelated files on different filesystems that happen to get the same inode number.

Reply:
Okay, I've added a check.

Reply:
we have 
https://github.com/uutils/coreutils/blob/8cac375dddb4d4e87519c137c45fa831ac7f2619/src/uucore/src/lib/features/fs.rs#L559
it would not be a good candidate ?

Reply:
> we have https://github.com/uutils/coreutils/blob/8cac375dddb4d4e87519c137c45fa831ac7f2619/src/uucore/src/lib/features/fs.rs#L559 it would not be a good candidate ?

Thanks! This has now been changed to `uucore::fs::paths_refer_to_same_file` in https://github.com/uutils/findutils/pull/389/commits/f612c0da06859beed4aeab498f6195eafe8de7b2 and it works great!